applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp

```diff
@@ -942,6 +942,257 @@ Matrix ConstitutiveLawUtilities<TVoigtSize>::CalculateDirectPlasticDeformationGr
+        rAsInv.resize(VoigtSize, VoigtSize);
+    }
+    noalias(rAsInv) = ZeroMatrix(VoigtSize, VoigtSize);
+    const Vector &r_iso_aniso_yield_ratios = r_mat_props[ISOTROPIC_ANISOTROPIC_YIELD_RATIO];
+    KRATOS_ERROR_IF_NOT(r_iso_aniso_yield_ratios.size() == VoigtSize) << "The length of the ISOTROPIC_ANISOTROPIC_YIELD_RATIO is not correct" << std::endl;
+
+    if (VoigtSize == 6) {
+        rAs(0, 0) = r_iso_aniso_yield_ratios(0);
+        rAs(1, 1) = r_iso_aniso_yield_ratios(1);
+        rAs(2, 2) = r_iso_aniso_yield_ratios(2);
+        rAs(3, 3) = r_iso_aniso_yield_ratios(3);
+        rAs(4, 4) = r_iso_aniso_yield_ratios(4);
+        rAs(5, 5) = r_iso_aniso_yield_ratios(5);
+    } else {
+        rAs(0, 0) = r_iso_aniso_yield_ratios(0);
+        rAs(1, 1) = r_iso_aniso_yield_ratios(1);
+        rAs(2, 2) = r_iso_aniso_yield_ratios(2);
+    }
+    for (IndexType i = 0; i < VoigtSize; ++i)
+        rAsInv(i, i) = 1.0 / rAs(i, i);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TVoigtSize>
+void ConstitutiveLawUtilities<TVoigtSize>:: CalculateAnisotropicStrainMapperMatrix(
+    const Matrix& rAnisotropicElasticMatrix,
+    const Matrix& rIsotropicElasticMatrix,
+    const Matrix &rAs,
+    Matrix& rAe
+)
+{
+    if (rAe.size1() != VoigtSize || rAe.size2() != VoigtSize)
+        rAe.resize(VoigtSize, VoigtSize);
+    noalias(rAe) = ZeroMatrix(VoigtSize, VoigtSize);
+
+    Matrix inv_isotropic_elastic_matrix(VoigtSize, VoigtSize);
+    noalias(inv_isotropic_elastic_matrix) = ZeroMatrix(VoigtSize, VoigtSize);
+    double aux_det;
+    MathUtils<double>::InvertMatrix(rIsotropicElasticMatrix, inv_isotropic_elastic_matrix, aux_det);
+    noalias(rAe) = prod(inv_isotropic_elastic_matrix, Matrix(prod(rAs, rAnisotropicElasticMatrix)));
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void ConstitutiveLawUtilities<6>::CalculateRotationOperatorEuler1(
+    const double EulerAngle1,
+    Matrix& rRotationOperator
+)
+{
+    if (rRotationOperator.size1() != Dimension || rRotationOperator.size2() != Dimension)
+        rRotationOperator.resize(Dimension, Dimension);
+    noalias(rRotationOperator) = ZeroMatrix(Dimension, Dimension);
+
+    const double cos_angle = std::cos(EulerAngle1 * Globals::Pi / 180.0);
+    const double sin_angle = std::sin(EulerAngle1 * Globals::Pi / 180.0);
+
+    rRotationOperator(0, 0) = cos_angle;
+    rRotationOperator(0, 1) = sin_angle;
+    rRotationOperator(1, 0) = -sin_angle;
+    rRotationOperator(1, 1) = cos_angle;
+    rRotationOperator(2, 2) = 1.0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
```

Comment:
you need to specify the convention you adopt (order in which rotations are applied)